feat(datasource/adoptium): support jre only releases (#11652)
renovatebot/renovate: lib/datasource/adoptium-java/readme.md

This datasource returns releases from [Adoptium](https://adoptium.net/) API.

It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.

It only uses the first 50 pages with 50 items per page.

Use `java-jdk` or `java` as `lookupName` to get releases which come with the Java Development Kit.

> Use `java-jre`  as `lookupName` if you only want releases which come with the Java Runtime Environment.

Author: ```suggestion
Use `java-jre` as `lookupName` if you only want releases which come with the Java Runtime Environment.
```

This datasource returns releases from [Adoptium](https://adoptium.net/) API.

It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.

It only uses the first 50 pages with 50 items per page.

> Use `java-jdk` or `java` as `lookupName` to get releases with JDK available or use `java-jre` to only get releases with JRE available.

Reviewer: ```suggestion
Use `java-jdk` or `java` as `lookupName` to get releases which come with the Java Development Kit.

Use `java-jre`  as `lookupName` if you only want releases which come with the Java Runtime Environment.
```

Maybe something like this is easier to read and understand?

> Currently only the LTS releases have a JRE available.

Reviewer: ```suggestion
Currently only the LTS releases of Java come with the JRE.
```

I think we mean that only LTS releases of Java come with the Java Runtime Environment?
